### ISSUE TITLE
Wrong sha256 for ubuntu image

### ISSUE THREAD

**Issue by brunoalr** [PPC64/sha2-le#4]
The file http://cdimage.ubuntu.com/releases/16.10/release/ubuntu-16.10-server-ppc64el.iso should have SHA256 sum value of  `d14bdb413ea6cdc8d9354fcbc37a834b7de0c23f992deb0c6764d0fd5d65408e ` according to http://cdimage.ubuntu.com/releases/16.10/release/SHA256SUMS and the core-utils `sha256sum` command.

Instead, sha256 returns `aef33b9074cce6a7ffa4f9450c84b21a469bc987178edebc983c3a875eb89134`

**Comment by racardoso** [comment]
The problem is with `write_size` function. It's not write the size on the high 32 bits of the 64 bit block that contains the message length:

```
Breakpoint 1, write_size (input=0x3fff69010010 "", size=661370880, position=661370936) at sha2.h:103
...
(gdb) x total_size
0x3fff906cb84c:	0x3b5dc000
(gdb) x total_size-1
0x3fff906cb848:	0x00000000 // **should be 0x00000001**
```
Setting it manually give us the correct result.

```
(gdb) set {int}0x3fff906cb848=1
(gdb) x total_size-1
0x3fff906cb848:	0x00000001
(gdb) c
Continuing.
d14bdb413ea6cdc8d9354fcbc37a834b7de0c23f992deb0c6764d0fd5d65408e
```
I'll fix this function.

**Comment by racardoso** [comment]
Message length is `number of chars * sizeof(char)`. The parameter `size` only give us the number of chars in message and we need to multiply it by `sizeof(char)` before shift by 32. Closed via 92349bffbaae3b6836339644a5c177be0d5d6ac1

**Comment by gut** [comment]
Ops, my bad! Thanks
